tutorials/bert_e2e.py

```diff
@@ -0,0 +1,206 @@
+    real_outputs = session.run([i.name for i in session.get_outputs()], inputs)
+    matched_idx = [i for i, o in enumerate(session.get_outputs()) if list(ref_map_out.keys())[0] in o.name][0]
+
+    assert np.allclose(
+        real_outputs[matched_idx], ref_outputs[0], atol=1e-2,rtol=1e-12
+    ), f"Results do not match, expected:{ref_outputs[0]}, but got {real_outputs[matched_idx] }"
+    print("Results matches:", real_outputs[0], "\ndiff:", real_outputs[matched_idx] - ref_outputs[0])
+    
+
+def main():
+    parser = argparse.ArgumentParser(
+        os.path.basename(__file__),
+        description="""Add pre and post processing to a model.
+
+        This tutorial supports updating:
+          - MobileBert with different tasks
+          - XLM-Roberta with classification task   
+          
+        This script does only server as a example of how to add pre/post processing to a transformer model.
+        Usually pre-processing includes tokenizer and basic conversion of input_ids after tokenizer.
+        Post-processing includes conversion of output_ids to text.
+        
+        A pipeline is created for organizing the required pre/post processing 'Steps' in the order required. 
+        Configure individual steps as needed. 
+
+        The updated model will be written in the same location as the original model, with '.onnx' updated to 
+        '.with_pre_post_processing.onnx'.
+        
+        Export models from huggingface by default if there is not a onnx model provided.
+        NOTE: if provided a onnx model, you have to make sure your model is matched with the model_type in hugging-face.
+        As we are using the hugging-face tokenizer to do the pre-processing.
+        """,
+    )
+
+    parser.add_argument(
+        "-t",
+        "--model_type",
+        type=str,
+        required=True,
+        choices=[
+            "xlm-roberta-base",
+            "google/mobilebert-uncased",
+            "csarron/mobilebert-uncased-squad-v2",
+            "lordtt13/emo-mobilebert",
+        ],
+        help="Model type.",
+    )
+
+    parser.add_argument(
+        "model_path",
+        type=Path,
+        help="""The onnx model path which needs pre-or-post processing update 
+                        or a dir-path where to save updated-model.
+                        If a onnx model path is provided, the updated model will be written in the sane directory
+                        with a suffix '.with_pre_post_processing.onnx'""",
+    )
+
+    args = parser.parse_args()
+
+    model_path = args.model_path.resolve(strict=True)
+    canonized_name = re.sub(r"[^a-zA-Z0-9]", "_", args.model_type) + ".onnx"
+
+    if model_path.is_dir():
+        model_path = model_path / canonized_name
+
+    new_model_path = model_path.with_suffix(".with_pre_post_processing.onnx")
+
+    add_pre_post_processing_to_transformers(args.model_type, model_path, new_model_path)
+    verify_results_for_e2e_model(new_model_path, args.model_type, model_path)
```

Comment:
nit: Given we're calling this with all 3 args we don't need a default value for new_model_path do we? If not, could we make the order of args args.model_type, model_path, new_model_path?